third_party/coro/coro.c

```diff
@@ -320,21 +338,32 @@ trampoline (int sig)
          ".fnend\n"
 
        #elif __aarch64__
-
+#ifdef ENABLE_BACKTRACE
          ".cfi_startproc\n"
-         "\tmov x30, #0\n"
-         "\tsub sp, sp, #16\n"
-         "\tstr x30, [sp, #0]\n"
-         ".cfi_def_cfa_offset 16\n"
-         ".cfi_offset 30, -16\n"
+	 /*
+	  * Call-chain ends here: we need to invalidate this frame's return
+	  * address to make unwinding stop here.
+	  * No nuances: AARCH64 has a special link register for storing
+	  * return addresses.
+	  */
+	 ".cfi_undefined x30\n"
          "\tmov x0, x20\n"
          "\tblr x19\n"
-# ifdef __APPLE__
+#ifdef __APPLE__
          "\tb _abort\n"
-# else
+#else /* __APPLE__ */
          "\tb abort\n"
-# endif
+#endif /* __APPLE__ */
          ".cfi_endproc\n"
+#else /* ENABLE_BACKTRACE */
+	 "\tmov x0, x20\n"
+         "\tblr x19\n"
+#ifdef __APPLE__
+         "\tb _abort\n"
```

Comment:
It seem that there several lines identical in both branches `#ifdef ENABLE_BACKTRACE`. Could we make it outside of `#ifdef ENABLE_BACKTRACE`?